app/models/user.rb

```diff
@@ -1,5 +1,8 @@
 class User < ApplicationRecord
-  validates :name, presence: true, uniqueness: true
+  EMAIL_REGEX = /\A(|(([A-Za-z0-9]+_+)|([A-Za-z0-9]+\-+)|([A-Za-z0-9]+\.+)|([A-Za-z0-9]+\++))*[A-Za-z0-9]+@((\w+\-+)|(\w+\.))*\w{1,63}\.[a-zA-Z]{2,6})\z/i
+
+  validates :name, :email, presence: true
+  validates :email, uniqueness: { message: "already exists" }, format: { with: EMAIL_REGEX }
```

Comment:
If msg is generic, then we dont need to pass it explicitly.

Reply:
Made suggested changes